### ISSUE TITLE
futarchy-ts SDK Interface

### ISSUE THREAD

**Issue by R-K-H** [metaDAOproject/futarchy-sdk#2]
There are shared functionalities across multiple systems, there are scripts which exist in https://github.com/metaDAOproject/futarchy repo, aspects tightly coupled with the current https://github.com/metaDAOproject/meta-dao-frontend and aspects in our indexing.

We should resolve this code duplication and efforts such that these things reside in a singular repo.

Given that this is an evolving aspect of our need (and logical) the goal here would be to bring scripting and interactions with the programs under one roof.

Where this line is drawn currently is a bit blurry, but it's likely fine to favor more in here than less.

Some core pieces would be:
- [x] Types / Structuring
- [x] Fetching Autocract Data
- [x] Fetching DAO Data
- [x] Fetch TWAP Oracle Data
- [x] Initialize Proposal
- [x] Finalize Proposal
- [ ] Initialize DAO
- [x] Reconstitute Conditional Tokens 

Somethings which should be considered:
- Cranking
- Reclaim OpenBookV2 Rent
- OpenBookV2 Data Interfaces
- Transaction Manager
- State Manager

**Comment by decartesol** [comment]
Hopefully I can get time to focus on this work this week.

**Comment by LukasDeco** [comment]
@decartesol because I am working on https://github.com/metaDAOproject/futarchy-ts/issues/3 right now so this means I am working on the functionality to fetch dao data via the SDK. We might need to discuss SDK design that we can all agree on or if you are okay with me running with it that's fine too. Just wanting to unblock us.

**Comment by R-K-H** [comment]
@R-K-H split this out better

**Comment by decartesol** [comment]
@LukasDeco thanks for #3. I can try building on top of it

**Comment by decartesol** [comment]
Ultimately I think it's best for the frontend devs to have final say on the design since they know what they need best as consumers of the SDK

**Comment by LukasDeco** [comment]
@decartesol we should probably at some point walk though the types that have been created on the frontend and make sure that our postgres schema can support the kinds of joins and queries that seem to be warranted. cc @R-K-H

**Comment by R-K-H** [comment]
Adding this in as reference for a key component https://github.com/orgs/metaDAOproject/projects/11/views/1?pane=issue&itemId=59711443

**Comment by R-K-H** [comment]
The initialize DAO is supported in https://github.com/metaDAOproject/futarchy/blob/5d193faacfbc59c0e51230cfe950fb62acd1f9dd/sdk/src/AutocratClient.ts#L189

As well for review
https://github.com/metaDAOproject/futarchy/blob/5d193faacfbc59c0e51230cfe950fb62acd1f9dd/tests/autocrat.ts#L170
